### PR TITLE
Same order and field in DataRecord and Datatable

### LINKED ISSUE #32: [View] When editing a view show only the fields of the current view

### PR DESCRIPTION
**Pull request by Alketta**
Closes #32 

 - Add doc for each method in DatabaseTable

### CODE REVIEW

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue [hunk 1]
```diff
@@ -613,6 +686,10 @@ export default {
       this.multipleAutocompleteInput = {}
       this.displayNewDialog = true
     },
+    /**
+     * Export current view according to current filters in CSV
+     * Todo: Do the same for sort!
```
Comment (Alketta) [inline front/src/views/routes/workspace/admin/database/DatabaseTable.vue:691]: Todo: Export of the current view according to the sort if the user applies one.

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue [hunk 2]
```diff
@@ -392,30 +415,37 @@ export default {
       })
       return result
     },
-    editableColumns () {
-      if (!this.block.definition.columns) return []
-      return this.block.definition.columns.filter(c => this.isEditableColumn(this.crudMode, c))
-    },
+    /**
+     * Get the current view with table_actions and columns
```
Comment (mdartic) [inline front/src/views/routes/workspace/admin/database/DatabaseTable.vue:419]: I would prefer not use verb to describe the function, as it's a computed property, and not a method.
It returns something, but, this is not a "get" or "retrieve", I think ?

```suggestion
     * Current view with table_actions and columns
```

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue [hunk 3]
```diff
@@ -392,30 +415,37 @@ export default {
...
       if (!this.selectedViewId) return null
       return this.views.find(({ id }) => this.selectedViewId === id)
     },
-    displayColumnsView () {
-      if (!this.currentView) return { columns: [] }
-      const columns = this.currentView.columns.slice(0).sort((a, b) => a.position - b.position)
-      return { columns }
-    },
+    /**
+     * Retrieve all ids of the current sorted view
```
Comment (mdartic) [inline front/src/views/routes/workspace/admin/database/DatabaseTable.vue:427]:
```suggestion
     * Columns ids of current view
```

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue [hunk 4]
```diff
@@ -361,24 +361,47 @@ export default {
...
+      if (!this.currentView?.columns) return []
       return {
-        columns: this.block.definition.columns.filter((column) =>
+        columns: this.currentView.columns.filter((column) =>
           !READ_ONLY_COLUMNS_TYPES.has(column.column_type_id),
         ),
       }
     },
+    /**
+     * Sort columns according to their position in the current view
```
Comment (mdartic) [inline front/src/views/routes/workspace/admin/database/DatabaseTable.vue:388]:
```suggestion
     * Sorted columns according to their position in the current view
```

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue [hunk 5]
```diff
@@ -361,24 +361,47 @@ export default {
...
+     * In DataTable and DataRecord
+     * @returns {{columns: *}|{columns: *[]}}
+     */
+    displayColumnsView () {
+      if (!this.currentView) return { columns: [] }
+      const columns = this.currentView.columns.slice(0).sort((a, b) => a.position - b.position)
+      return { columns }
+    },
+    /**
+     * Retrieve all option information for each SINGLE_SELECT and MULTI_SELECT's column on the current view
```
Comment (mdartic) [inline front/src/views/routes/workspace/admin/database/DatabaseTable.vue:398]:
```suggestion
     * Options informations for each SINGLE_SELECT and MULTI_SELECT's column on the current view
```

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue [hunk 6]
```diff
@@ -392,30 +415,37 @@ export default {
...
     },
+    /**
+     * Check if data or column available in the current view
+     * @returns {boolean}
+     */
     hasDataToDisplay () {
       return this.displayColumnsView.columns.length > 0 && this.block?.content?.total > 0
     },
+    /**
+     * Transform in object where key = id and value = data
```
Comment (mdartic) [inline front/src/views/routes/workspace/admin/database/DatabaseTable.vue:442]:
```suggestion
     * Columns of current view transformed in object where key = id and value = data
```

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue [hunk 7]
```diff
@@ -494,27 +537,39 @@ export default {
...
+    /**
+     * Retrieve view with action and column
+     */
     async loadTableAndProcess () {
       this.block.loading = true
       this.block.content = {
         total: 0,
         data: null,
       }
+      // Legacy: doesn't use anymore
```
Comment (mdartic) [inline front/src/views/routes/workspace/admin/database/DatabaseTable.vue:556]: do you think we need to add a `TODO: ` ?

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue (hunk 1 above)
Comment (mdartic) [reply front/src/views/routes/workspace/admin/database/DatabaseTable.vue:691]: Maybe we need to create a new issue for this use case ?

**Review by mdartic — CHANGES_REQUESTED** [review]
Little changes, for your comments mainly.

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue (hunk 1 above)
Comment (Alketta) [reply front/src/views/routes/workspace/admin/database/DatabaseTable.vue:691]: Already done

File: front/src/views/routes/workspace/admin/database/DatabaseTable.vue (hunk 7 above)
Comment (Alketta) [reply front/src/views/routes/workspace/admin/database/DatabaseTable.vue:556]: I think we will remove it in the new version. So I don't think a "Todo" it's necessary.

**Review by mdartic — APPROVED** [review]

**Pull request merged**